File: main/src/main/java/com/sedmelluq/discord/lavaplayer/source/soundcloud/SoundCloudHtmlDataLoader.java
package com.sedmelluq.discord.lavaplayer.source.soundcloud;

import com.sedmelluq.discord.lavaplayer.tools.JsonBrowser;
import com.sedmelluq.discord.lavaplayer.tools.io.HttpInterface;
import java.io.IOException;

public interface SoundCloudHtmlDataLoader {
  JsonBrowser load(HttpInterface httpInterface, String url) throws IOException;
}

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/source/soundcloud/SoundCloudHtmlDataLoader.kt
package com.sedmelluq.discord.lavaplayer.source.soundcloud

import com.sedmelluq.discord.lavaplayer.tools.io.HttpInterface
import java.io.IOException

interface SoundCloudHtmlDataLoader {
    @Throws(IOException::class)
    fun load(httpInterface: HttpInterface, url: String): SoundCloudRootDataModel?
}
